--- a/src/tasks.py
+++ b/src/tasks.py
@@ -491,6 +491,8 @@ def install(c, uv=False, skip_plugins=False):
     INSTALL_FILE = local_dir().joinpath('src/backend/requirements.txt')

     info(f"Installing required python packages from '{INSTALL_FILE}'")
+    warning("Skipping install on NixOS")
+    return

     if not Path(INSTALL_FILE).is_file():
         raise FileNotFoundError(f"Requirements file '{INSTALL_FILE}' not found")
@@ -608,6 +610,8 @@ def remove_mfa(c, mail=''):
 @state_logger('TASK08')
 def static(c, frontend=False, clear=True, skip_plugins=False):
     """Copies required static files to the STATIC_ROOT directory, as per Django requirements."""
+    warning("Skipping static file generation on NixOS")
+    return
     if frontend and node_available():
         frontend_compile(c)

@@ -1541,6 +1545,8 @@ def frontend_compile(c, extract: bool = False):
         extract (bool): Whether to extract translations from source code. Defaults to False.
     """
     info('Compiling frontend code...')
+    warning("Skipping frontend compile on NixOS")
+    return
     frontend_install(c)
     frontend_trans(c, extract=extract)
     frontend_build(c)
@@ -1651,6 +1657,8 @@ def frontend_download(
     import requests

     info('Downloading frontend...')
+    warning("Skipping frontend download on NixOS")
+    return

     # globals
     default_headers = {'Accept': 'application/vnd.github.v3+json'}
