--- a/src/tasks.py
+++ b/src/tasks.py
@@ -233,6 +233,8 @@ def install(c, uv=False):
     INSTALL_FILE = 'src/backend/requirements.txt'
 
     print(f"Installing required python packages from '{INSTALL_FILE}'")
+    print(f"Skipping install on NixOS")
+    return
 
     if not Path(INSTALL_FILE).is_file():
         raise FileNotFoundError(f"Requirements file '{INSTALL_FILE}' not found")
@@ -319,6 +321,8 @@ def remove_mfa(c, mail=''):
 @task(help={'frontend': 'Build the frontend'})
 def static(c, frontend=False):
     """Copies required static files to the STATIC_ROOT directory, as per Django requirements."""
+    print("Skipping static file generation on NixOS")
+    return
     manage(c, 'prerender')
 
     if frontend and node_available():
@@ -335,6 +339,8 @@ def translate_stats(c):
 
     The file generated from this is needed for the UI.
     """
+    print("Skipping translating stats on NixOS")
+    return
     # Recompile the translation files (.mo)
     # We do not run 'invoke translate' here, as that will touch the source (.po) files too!
     try:
@@ -1076,6 +1082,8 @@ def frontend_compile(c):
         c: Context variable
     """
     print('Compiling frontend code...')
+    print("Skipping frontend compile on NixOS")
+    return
 
     frontend_install(c)
     frontend_trans(c)
@@ -1168,6 +1176,8 @@ def frontend_download(
     import requests
 
     print('Downloading frontend...')
+    print("Skipping frontend download on NixOS")
+    return
 
     # globals
     default_headers = {'Accept': 'application/vnd.github.v3+json'}
