diff --git a/src/backend/InvenTree/plugin/installer.py b/src/backend/InvenTree/plugin/installer.py
index b1fb2de03..d633a0adc 100644
--- a/src/src/backend/InvenTree/plugin/installer.py
+++ b/src/src/backend/InvenTree/plugin/installer.py
@@ -1,5 +1,6 @@
 """Install a plugin into the python virtual environment."""
 
+import os
 import re
 import subprocess
 import sys
@@ -13,11 +14,12 @@ import structlog
 import plugin.models
 import plugin.staticfiles
 from InvenTree.exceptions import log_error
+from InvenTree.config import get_plugin_dir
 
 logger = structlog.get_logger('inventree')
 
 
-def pip_command(*args):
+def pip_command(*args, env: dict[str, str] | None = None):
     """Build and run a pip command using using the current python executable.
 
     Returns: The output of the pip command
@@ -25,6 +27,10 @@ def pip_command(*args):
     Raises:
         subprocess.CalledProcessError: If the pip command fails
     """
+    if env is not None:
+        logger.warning('Running pip command with extra env vars: %s', str(env))
+        env = os.environ.copy() | env
+
     python = sys.executable
 
     command = [python, '-m', 'pip']
@@ -36,7 +42,7 @@ def pip_command(*args):
     logger.info('Running pip command: %s', ' '.join(command))
 
     return subprocess.check_output(
-        command, cwd=settings.BASE_DIR.parent, stderr=subprocess.STDOUT
+        command, cwd=settings.BASE_DIR.parent, stderr=subprocess.STDOUT, env=env
     )
 
 
@@ -80,9 +86,32 @@ def get_install_info(packagename: str) -> dict:
 
     info = {}
 
+    result = None
+
     try:
         result = pip_command('show', packagename)
+    except subprocess.CalledProcessError as error_base:
+        err_output = error_base.output.decode('utf-8')
+        logger.warning('Plugin lookup failed in base venv: %s', err_output)
+        info['error'] = err_output
+
+        plugin_dir = get_plugin_dir()
+        if plugin_dir:
+            logger.info('Looking for plugin in external plugin dir %s', plugin_dir)
+            try:
+                result = pip_command(
+                    'show', packagename, env={'PYTHONPATH': plugin_dir}
+                )
+            except subprocess.CalledProcessError as error_ext:
+                err_output = error_ext.output.decode('utf-8')
+                logger.warning(
+                    'Plugin lookup failed in external plugin dir: %s', err_output
+                )
+                info['error'] += err_output
+    except Exception:
+        log_error('get_install_info', scope='pip')
 
+    if result is not None:
         output = result.decode('utf-8').split('\n')
 
         for line in output:
@@ -93,15 +122,9 @@ def get_install_info(packagename: str) -> dict:
                 value = str(parts[1].strip())
 
                 info[key] = value
-
-    except subprocess.CalledProcessError as error:
-        log_error('get_install_info', scope='pip')
-
-        output = error.output.decode('utf-8')
-        info['error'] = output
-        logger.exception('Plugin lookup failed: %s', output)
-    except Exception:
+    else:
         log_error('get_install_info', scope='pip')
+        logger.exception('Plugin lookup failed: %s', info['error'])
 
     return info
 
@@ -134,7 +157,14 @@ def install_plugins_file():
         logger.warning('Plugin file %s does not exist', pf)
         return
 
-    cmd = ['install', '--disable-pip-version-check', '-U', '-r', str(pf)]
+    cmd = ['install', '--disable-pip-version-check', '-U']
+
+    plugin_dir = get_plugin_dir()
+    if plugin_dir:
+        logger.info('Installing into external plugin dir %s', plugin_dir)
+        cmd += ['--target', plugin_dir]
+
+    cmd += ['-r', str(pf)]
 
     try:
         pip_command(*cmd)
@@ -247,6 +277,11 @@ def install_plugin(url=None, packagename=None, user=None, version=None):
     # build up the command
     install_name = ['install', '-U', '--disable-pip-version-check']
 
+    plugin_dir = get_plugin_dir()
+    if plugin_dir:
+        logger.info('Installing into external plugin dir %s', plugin_dir)
+        install_name += ['--target', plugin_dir]
+
     full_pkg = ''
 
     if url:
