--- a/inventree-src/src/backend/InvenTree/InvenTree/settings.py
+++ b/inventree-src/src/backend/InvenTree/InvenTree/settings.py
@@ -218,6 +218,7 @@ INSTALLED_APPS = [
     'rest_framework',  # DRF (Django Rest Framework)
     'corsheaders',  # Cross-origin Resource Sharing for DRF
     'crispy_forms',  # Improved form rendering
+    'crispy_bootstrap4',  # https://stackoverflow.com/questions/75495403/django-returns-templatedoesnotexist-when-using-crispy-forms
     'import_export',  # Import / export tables to file
     'django_cleanup.apps.CleanupConfig',  # Automatically delete orphaned MEDIA files
     'mptt',  # Modified Preorder Tree Traversal
